scheduler/pkg/kafka/pipeline/kafkamanager.go

```diff
@@ -202,7 +202,7 @@ func (km *KafkaManager) Infer(
 		km.mu.RUnlock()
 		return nil, err
 	}
-	key := requestId
+	key := getCompositeKey(resourceName, requestId, ".")
```

Comment:
nit: just for legibility could you rename `key` to `compositeKey` and also add a note why we are doing it?